### PR TITLE
Add authentication routes and controllers and include refresh and access token

### PR DESCRIPTION
**Pull request by carpecodeum**
# Description
Add authentication routes and controllers and include refresh and access token including revokedtokens table and many to many changes in image and labels table. Also changed the directory structure.

Fixes #424 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
Screenshot 2020-06-01 at 7 54 04 PM

Also, include screenshots of app for the verification and reviewing purpose.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by carpecodeum** [comment]
@Tushar19varshney @ivantha @UdeshUK review this please.

File: labellab-flask/requirements.txt [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
 MarkupSafe==1.1.1
+pep8==1.7.1
+PyMySQL==0.9.3
+python-dateutil==2.8.1
 python-dotenv==0.11.0
+python-editor==1.0.4
 six==1.14.0
+SQLAlchemy==1.3.17
+typing==3.7.4.1
 Werkzeug==1.0.0
```
Comment (UdeshUK) [inline labellab-flask/requirements.txt:21]: Should add ```Flask-JWT-Extended```, ```Flask-Bcrypt``` dependencies

File: labellab-flask/api/controllers/userscontroller.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    """This class registers a new user."""
+
+    def post(self):
+        """Handle POST request for this view. Url --> /api/users/register"""
+        # getting JSON data from request
+        post_data = request.get_json(silent=True,force=True)
+
+        try:
+            name = post_data["name"]
+            username = post_data["username"]
```
Comment (UdeshUK) [inline labellab-flask/api/controllers/userscontroller.py:24]: Can we drop ```username``` attribute and use just the email? @ivantha @Tushar19varshney

Comment (carpecodeum) [reply labellab-flask/api/controllers/userscontroller.py:24]: okay sure we can remove it if others also agree

File: labellab-flask/requirements.txt (hunk 1 above)
Comment (carpecodeum) [reply labellab-flask/requirements.txt:21]: sure

File: labellab-flask/api/controllers/userscontroller.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+        # Querying the database with requested email
+        user = User.find_by_email(email)
+
+        if user:
+            # There is an existing user. We don't want to register users twice
+            # Return a message to the user telling them that they they already
+            # exist
+            response = {"message": "Email already exists. Please login."}
+            return make_response(jsonify(response)), 202
```
[the remark this answers is not in the document]
Comment (carpecodeum) [reply labellab-flask/api/controllers/userscontroller.py:40]: okay sure we can change it

File: labellab-flask/api/controllers/userscontroller.py (hunk 2 above)
Comment (UdeshUK) [reply labellab-flask/api/controllers/userscontroller.py:24]: I'll merge this for now. We'll remove it later if it is redundant.

**Pull request merged**